### ISSUE TITLE
Refactor `forc-client` to be aware of workspace manifest files

### ISSUE THREAD

**Issue by kayagokalp** [FuelLabs/sway#2929]
After #2893 is addressed we are going to have a workspace manifest file that holds various information for that workspace. As a first step going towards full workspace support we may want to do some refactoring around `forc-client` so that it can be aware of the workspace manifest file. Until we add the deployment dependencies to the workspace manifest file, we can deploy the members of the workspace as they are declared.

**Comment by mitchmindtree** [comment]
> Until we add the deployment dependencies to the workspace manifest file

Just a note to add that we'll want to do this in the package manifest (not the workspace manifest) as mentioned here https://github.com/FuelLabs/sway/issues/2931#issuecomment-1267775595.

@kayagokalp we might be better off tackling #2931 first and leave any necessary refactoring of `forc-client` until after this. If we end up outputting the contract ID during `forc build` instead as mentioned [here](https://github.com/FuelLabs/sway/issues/2930#issuecomment-1267800047), then we shouldn't need to do much in `forc-client` at all other than:

1. Refactor `deploy` and `run` to use the contract ID from the `out` directory (or directly from the `forc_pkg` build function call used within the `deploy` function).
2. Add support for `deploy`ing all contract packages within a workspace at once.

**Comment by kayagokalp** [comment]
Ahh just realized this exists after I created #3156. Since we longer have 

> Refactor deploy and run to use the contract ID from the out directory (or directly from the forc_pkg build function call used within the deploy function).

section of your summary as we handled that with introducing `contract-dependencies` at https://github.com/FuelLabs/sway/pull/3016. And second part of this issue is tracked by newer smaller issue #3156. I am closing this one but if you feel like tracking with this makes more sense I can reverse
